pkg/cloudfabric-controller/controllerframework/controller_framework_test.go

```diff
@@ -290,7 +292,460 @@ func TestConsolidateControllerInstances_ReturnValues_MergeAndAutoExtends(t *test
 	assert.Equal(t, int64(math.MaxInt64), controllerInstanceBase.sortedControllerInstancesLocal[0].controllerKey)
 }
 
-func TestGetMaxInterval(t *testing.T) {
+// This test is to check while keep adding new controller manager instance, the scope will be evenly distributed
+func TestGenerateKeyContinuously(t *testing.T) {
+	controllerBase := new(ControllerBase)
+	controllerBase.sortedControllerInstancesLocal = []controllerInstanceLocal{}
+
+	// Totally add 1000 controller manager instance one by one
+	const TotalInstanceCount = 1000
+	for i := 0; i < TotalInstanceCount; i++ {
+		controllerKey, _ := controllerBase.generateKey()
+		controllerInstanceLocal := new(controllerInstanceLocal)
+		controllerInstanceLocal.controllerKey = controllerKey
+		generateTestSortedControllerInstances(controllerBase, controllerInstanceLocal)
+		verifyControllerKeyEvenlyDistributed(t, controllerBase)
+	}
+}
+
+func generateTestSortedControllerInstances(controllerBase *ControllerBase, controllerInstanceLocal *controllerInstanceLocal) {
+	// Generate controllerInstanceMap
+	controllerInstanceMap := make(map[string]v1.ControllerInstance)
+	index := 0
+	for _, instance := range controllerBase.sortedControllerInstancesLocal {
+		mapInstance := v1.ControllerInstance{
+			ControllerKey: instance.controllerKey,
+		}
+		controllerInstanceMap[string(index)] = mapInstance
```

Comment:
Why not use int as key type?

Reply:
I'm borrowing existing fun SortControllerInstancesByKeyAndConvertToLocal which key type is string.

Reply:
It does not justify